### PR TITLE
Cleanup `SimdPolicy` avoid messing with underlying `TeamPolicy`

### PR DESCRIPTION
**Pull request by dalg24**
Do not mess with the properties of the `Kokkos::TeamPolicy`, let the functor accept non-void work tags as leading argument.

### CODE REVIEW

**Comment by dalg24** [comment]
This is ready for review.  I surely can't be blamed here for the coverage.

File: core/src/Cabana_Parallel.hpp [hunk 1]
```diff
@@ -74,6 +74,54 @@ KOKKOS_FORCEINLINE_FUNCTION
...
+        else
+            Kokkos::parallel_for(
+                label, dynamic_cast<const team_policy&>( exec_policy_ ),
+                *this );
+    }
+
+    template <class WorkTag>
+    KOKKOS_FUNCTION std::enable_if_t<!std::is_void<WorkTag>::value &&
+                                     std::is_same<WorkTag, work_tag>::value>
+    operator()( WorkTag, member_type const& team ) const
```
Comment (streeve) [inline core/src/Cabana_Parallel.hpp:108]: So this is just adding flexibility if you want a different tag than the `SimdPolicy`?

**Review by streeve — APPROVED** [review]
This is great - likely the structure we want for updating the neighbor parallel

**Comment by streeve** [comment]
> I surely can't be blamed here for the coverage.

Agreed - opened #483 to remind us to add

File: core/src/Cabana_Parallel.hpp (hunk 1 above)
Comment (streeve) [reply core/src/Cabana_Parallel.hpp:108]: @dalg24 walked me through this - resolving

**Review by sslattery — APPROVED** [review]
Very nice! I like this solution. Much better way to do the tag dispatch.

**Comment by dalg24** [comment]
What's going on with the check statuses?  It looks like they are some pending duplicates.
What is the policy?  Can we override and merge?

**Comment by streeve** [comment]
It's because #444 updated the required tests. It's easier to rebase and rerun - otherwise we have to manually enable/disable all the tests again

**Pull request merged**